### PR TITLE
Update ForwardIndexType for removal of ++ operator

### PR DESCRIPTION
**Pull request by nielsandriesse**

### CODE REVIEW

**Comment by nielsandriesse** [comment]
The updated version of _advanceForward(_:) shows 4 more instructions in the SIL output, but I don't think we can get around that: https://gist.github.com/nielsandriesse/b065485f6c4756337a7e

File: stdlib/public/core/Index.swift [hunk 1]
```diff
@@ -214,9 +214,11 @@ extension ForwardIndexType {
     _precondition(n >= 0,
         "Only BidirectionalIndexType can be advanced by a negative amount")
     var p = self
-    for var i: Distance = 0; i != n; ++i {
-      ++p
-    }
+    var i: Distance = 0
+    repeat {
+      p._successorInPlace()
```
Comment (manoffewwords) [inline stdlib/public/core/Index.swift]: Just*.! On spa at free wire

**Comment by nielsandriesse** [comment]
Comparison of SIL: https://gist.github.com/nielsandriesse/b065485f6c4756337a7e . The SIL of the new version appears like it might be slightly better because the witness methods are loaded before the loop instead of in it.

**Comment by nielsandriesse** [comment]
This is basically the same case as _advanceForward(_:)

**Comment by nielsandriesse** [comment]
SIL is identical

**Pull request closed without merging**